pkg/app/config.go

```diff
@@ -487,11 +489,23 @@ func MergeAzureUserArgs(config Config, azureArgs *AzureArgs) error {
 		return fmt.Errorf("failed to set %s: %w", PackerAzureGalleryLocations, err)
 	}
 
+	// The gallery_name must be unique and represented by unique Azure publisher, offer, sku combination
+	// Together these values creates URN that represents a image in the gallery. example: Publisher:Offer:Sku:Version
+	// https://learn.microsoft.com/en-us/azure/virtual-machines/windows/cli-ps-findimage#terminology
+	// Create unique gallery image name to represent different OS flavors + kubernetes version + fips + metadata(release)
+	// to prevent conflicts when creating azure images.
+	// ex. dkp-ubuntu-2004-release-1.24.6-fips.0,  dkp-ubuntu-2004-release-1.24.6-nvidia
+	fullKuberenetesVersion, err := config.GetWithError(KubernetesFullVersionKey)
+	if err != nil {
+		return fmt.Errorf("unable to get full kubernetes version from config: %w", err)
+	}
 	galleryImageName := azureArgs.GalleryImageName
 	if galleryImageName == "" {
-		galleryImageName = fmt.Sprintf("dkp-%s", BuildName(config))
+		galleryImageName = fmt.Sprintf("dkp-%s-%s", BuildName(config), fullKuberenetesVersion)
 	}
 
+	galleryRegex := regexp.MustCompile(azureaGlleryImageNameRegex)
+	galleryImageName = galleryRegex.ReplaceAllString(galleryImageName, "-")
```

Comment:
only alphabets, numbers, `. _ -` allowed.  `fips+0` breaks this pattern. so sanitizing the gallery image name.